keps/672-execution-policy/README.md

```diff
@@ -0,0 +1,401 @@
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a KEP and for
+highlighting any additional information provided beyond the standard KEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API Details](#api-details)
+  - [Implementation](#implementation)
+  - [Quota Management](#quota-management)
+  - [Defaulting/Validation](#defaultingvalidation)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+  - [Manage quota for Job sequence](#manage-quota-for-job-sequence)
+  - [Support complex DAGs with JobSet](#support-complex-dags-with-jobset)
+- [Alternatives](#alternatives)
+  - [Add ExecutionPolicyRule parameter into the StartupPolicy API](#add-executionpolicyrule-parameter-into-the-startuppolicy-api)
+  <!-- /toc -->
+
+## Summary
+
+This KEP outlines the proposal to add the ExecutionPolicy API into JobSet. The ExecutionPolicy API
+allows to run sequence of ReplicatedJobs within a single JobSet.
+
+## Motivation
+
+Currently, JobSet supports the StartupPolicy API which allows to create Jobs in order after the
+first Job is in ready status. This can be useful when driver should be ready before workers.
+However, sometime high performance computing and machine learning users want to run sequence of
+ReplicatedJobs within JobSet. For example, it is common to run pre-processing,
+distributed fine-tuning, post-processing for LLM fine-tuning.
+
+### Goals
+
+- Add the ExecutionPolicy API for JobSet
+
+### Non-Goals
+
+- Support workflow management like DAGs outside of Job sequence.
+  - Users should consider to use Argo Workflows or Tekton Pipelines for such use-cases.
+- Allowing for a percentage of Jobs in a ReplicatedJob to be ready to consider the
+  whole ReplicatedJob to ready.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user, I want to fine-tune LLM using multi-node training with JobSet. I have the first
+ReplicatedJob for pre-trained model and dataset initialization, and the second ReplicatedJob for
+distributed fine-tuning.
+
+The example of JobSet looks as follows:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: fine-tune-llm
+spec:
+  executionPolicy:
+    executionPolicyOrder: InOrder
+    rules:
+      - targetReplicatedJobs:
+          - initializer
+        replicatedJobsStatus: Succeeded
```

Comment:
As a suggestion, I think naming this field `waitForReplicatedJobsStatus` would make the semantics of this clearer.